improver/nbhood/recursive_filter.py

```diff
@@ -226,81 +242,87 @@ def _run_recursion(cube, alphas_x, alphas_y, iterations):
-                                                      x_index)
-            output = RecursiveFilter._recurse_backward(output, alphas_x.data,
-                                                       x_index)
-            output = RecursiveFilter._recurse_forward(output, alphas_y.data,
-                                                      y_index)
-            output = RecursiveFilter._recurse_backward(output, alphas_y.data,
-                                                       y_index)
+            output = RecursiveFilter._recurse_forward(
+                output, smoothing_coefficients_x.data, x_index)
+            output = RecursiveFilter._recurse_backward(
+                output, smoothing_coefficients_x.data, x_index)
+            output = RecursiveFilter._recurse_forward(
+                output, smoothing_coefficients_y.data, y_index)
+            output = RecursiveFilter._recurse_backward(
+                output, smoothing_coefficients_y.data, y_index)
             cube.data = output
         return cube
 
-    def _set_alphas(self, cube, alpha, alphas_cube):
+    def _set_smoothing_coefficients(self, cube, smoothing_coefficient,
+                                    smoothing_coefficients_cube):
         """
-        Set up the alpha parameter.
+        Set up the smoothing_coefficient parameter.
 
         Args:
             cube (iris.cube.Cube):
                 2D cube containing the input data to which the recursive
                 filter will be applied.
-            alpha (float):
+            smoothing_coefficient (float):
                 The constant used to weight the recursive filter in that
-                direction: Defined such that 0.0 < alpha < 1.0
-            alphas_cube (iris.cube.Cube or None):
-                Cube containing array of alpha values that will be used
-                when applying the recursive filter in a specific direction.
-
+                direction: Defined such that 0.0 < smoothing_coefficient < 1.0
+            smoothing_coefficients_cube (iris.cube.Cube or None):
+                Cube containing array of smoothing_coefficient values that will
+                be used when applying the recursive filter in a specific
+                direction.
         Raises:
-            ValueError: If both alphas_cube and alpha are provided.
-            ValueError: If alpha and alphas_cube are both set to None
-            ValueError: If dimension of alphas array is less than dimension
-                        of data array
-            ValueError: If dimension of alphas array is greater than dimension
-                        of data array
+            ValueError: If both smoothing_coefficients_cube and
+                        smoothing_coefficient are provided.
+            ValueError: If smoothing_coefficient and
+                        smoothing_coefficients_cube are both set to None
+            ValueError: If dimension of smoothing_coefficients array is less
```

Comment:
Can't these two be combined into one thing, saying if dimensions don't match

Reply:
Done.